File: Java/BubbleSort.java
import java.util.Arrays;

public class BubbleSort {
    //Bubble sort,is a simple sorting algorithm that repeatedly steps through the list, compares adjacent elements and swaps them if they are in the wrong order.
    // The pass through the list is repeated until the list is sorted.
    //Time Complexity:
    // Best Case :O(n) ,Average Case:	O(n2) ,Worst Case	:O(n2)
    //Alorithm Explaination:https://www.javatpoint.com/bubble-sort
    public static void main(String[] args) {
        int[] arr={ 3, 6, 1, 8 ,4};
        bubble(arr);
        System.out.println(Arrays.toString(arr));
    }

    static int[] bubble(int[] arr) {
        //iterate n-1 times
        for (int i = 0; i < arr.length; i++) {
            //checking if adjacent right element is less than current element
            for (int j = i+1; j < arr.length; j++) {
                if(arr[j]<arr[i]){
                    //swapping the lesser element to the left position
                    int temp=arr[i];
                    arr[i]=arr[j];
                    arr[j]=temp;
                }
            }
        }
    return arr;
    }

}

File: Python/BubbleSort.py
def bubblesort(list1):

	#loop to access each element in the list
	for i in range (len(list1)):

		#loop to compare between two elements and go through the entire list
		for j in range (0,len(list1)-1):

			#comparing two adjacent elements
			if list1[j]>list1[j+1]:
        
				#swapping the two elements using a temp variable
				tempVariable = list1[j]
				list1[j] = list1 [j+1]
				list1[j+1] = tempVariable

data = [5,21,33,11,9,31]
bubblesort(data)
print (data)


